entityx/help/UIDGenerator.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <type_traits>
+
+namespace entityx {
+
+class UIDGenerator final {
+ public:
+  typedef std::size_t Family;
+
+  UIDGenerator(void) = delete;
+  UIDGenerator(const UIDGenerator &) = delete;
+  UIDGenerator &operator=(const UIDGenerator &) = delete;
+  ~UIDGenerator(void) = delete;
+
+  template <typename T>
+  static typename std::enable_if<!std::is_const<T>::value && !std::is_volatile<T>::value, std::size_t>::type GetUID(void) {
```

Comment:
Same comment as above with naming conventions: `get_uid()`, `get_counter_and_increment()`, and so on.